### PR TITLE
Cleanup codebase

### PR DESCRIPTION
**Pull request by Ticki**

### CODE REVIEW

File: filesystem/apps/zfs/metaslab.rs [hunk 1]
```diff
@@ -5,82 +5,82 @@ pub struct MetaslabClass;
...
+// segments are represented in-core by the ms_tree and are stored in an
+// AVL tree.
+//
+// As the space map grows (as a result of the appends) it will
+// eventually become space-inefficient. When the metaslab's in-core free tree
+// is zfs_condense_pct/100 times the size of the minimal on-disk
+// representation, we rewrite it in its minimized form. If a metaslab
+// needs to condense then we must set the ms_condensing flag to ensure
+// that allocations are not performed on the metaslab that is being written.
+//
```
Comment (stratact) [inline filesystem/apps/zfs/metaslab.rs]: This looks like it would be good for doc comments?

File: filesystem/apps/zfs/metaslab.rs [hunk 2]
```diff
@@ -91,17 +91,15 @@ pub struct Metaslab {
...
-     * same number of segments as the ms_tree. The only difference
-     * is that the ms_size_tree is ordered by segment sizes.
-     */
+    //
+    // The metaslab block allocators can optionally use a size-ordered
+    // range tree and/or an array of LBAs. Not all allocators use
+    // this functionality. The ms_size_tree should always contain the
+    // same number of segments as the ms_tree. The only difference
+    // is that the ms_size_tree is ordered by segment sizes.
+    //
```
Comment (stratact) [inline filesystem/apps/zfs/metaslab.rs]: Same here.

File: filesystem/schemes/orbital/display.rs [hunk 3]
```diff
@@ -95,26 +95,26 @@ impl Display {
...
-    /* Optimized { */
+    // Optimized {
     pub unsafe fn set_run(data: u32, dst: usize, len: usize) {
         let mut i = 0;
-        /*
-        //Only use 16 byte transfer if possible
-        if len - (dst + i) % 16 >= mem::size_of::<u32x4>() {
-            //Align 16
-            while (dst + i) % 16 != 0 && len - i >= mem::size_of::<u32>() {
-                *((dst + i) as *mut u32) = data;
```
Comment (stratact) [inline filesystem/schemes/orbital/display.rs]: Erm, why did you remove the deref operator here?

File: filesystem/schemes/orbital/display.rs [hunk 4]
```diff
@@ -95,26 +95,26 @@ impl Display {
...
-                i += mem::size_of::<u32>();
-            }
-            //While 16 byte transfers
-            let simd: u32x4 = u32x4(data, data, data, data);
-            while len - i >= mem::size_of::<u32x4>() {
-                *((dst + i) as *mut u32x4) = simd;
```
Comment (stratact) [inline filesystem/schemes/orbital/display.rs]: Same here. Why remove it?

File: filesystem/schemes/orbital/display.rs [hunk 5]
```diff
@@ -123,22 +123,22 @@ impl Display {
 
     pub unsafe fn copy_run(src: usize, dst: usize, len: usize) {
         let mut i = 0;
-        /*
-        //Only use 16 byte transfer if possible
-        if (src + i) % 16 == (dst + i) % 16 {
-            //Align 16
-            while (dst + i) % 16 != 0 && len - i >= mem::size_of::<u32>() {
-                *((dst + i) as *mut u32) = *((src + i) as *const u32);
```
Comment (stratact) [inline filesystem/schemes/orbital/display.rs]: Same here.

File: filesystem/schemes/orbital/display.rs [hunk 6]
```diff
@@ -123,22 +123,22 @@ impl Display {
...
-                i += mem::size_of::<u32>();
-            }
-            //While 16 byte transfers
-            while len - i >= mem::size_of::<u32x4>() {
-                *((dst + i) as *mut u32x4) = *((src + i) as *const u32x4);
```
Comment (stratact) [inline filesystem/schemes/orbital/display.rs]: Same here.

File: kernel/common/get_slice.rs [hunk 7]
```diff
@@ -11,27 +11,36 @@ use core::str;
...
+pub trait GetSlice {
+    fn get_slice(&self, a: Option<usize>, b: Option<usize>) -> &Self; }
 
 impl GetSlice for str {
     fn get_slice(&self, a: Option<usize>, b: Option<usize>) -> &Self {
         let slice = unsafe { slice::from_raw_parts(self.as_ptr(), self.len()) };
         let a = if let Some(tmp) = a {
             let len = slice.len();
-            if tmp > len { len }
-            else { tmp }
```
Comment (stratact) [inline kernel/common/get_slice.rs]: I actually like keeping these if-else expressions as one-liners here. Is there some reason to need to newline the braces for just one statement?

File: kernel/graphics/display.rs [hunk 8]
```diff
@@ -136,21 +136,21 @@ impl Display {
...
-            while len - i >= mem::size_of::<u32x4>() {
-                *((dst + i) as *mut u32x4) = *((src + i) as *const u32x4);
-                i += mem::size_of::<u32x4>();
-            }
-        }
-        */
+        // if (src + i) % 16 == (dst + i) % 16 {
+        // Align 16
+        // while (dst + i) % 16 != 0 && len - i >= mem::size_of::<u32>() {
+        // ((dst + i) as *mut u32) = *((src + i) as *const u32);
```
Comment (stratact) [inline kernel/graphics/display.rs]: Again, the deref operator `*` has been removed here.

File: kernel/graphics/display.rs [hunk 9]
```diff
@@ -136,21 +136,21 @@ impl Display {
...
+        // i += mem::size_of::<u32>();
+        // }
+        // While 16 byte transfers
+        // while len - i >= mem::size_of::<u32x4>() {
+        // ((dst + i) as *mut u32x4) = *((src + i) as *const u32x4);
```
Comment (stratact) [inline kernel/graphics/display.rs]: and here.

File: kernel/syscall/common.rs [hunk 10]
```diff
@@ -5,13 +5,13 @@ pub const SYS_BRK: usize = 45;
 pub const SYS_CHDIR: usize = 12;
 pub const SYS_CLOSE: usize = 6;
 pub const SYS_CLONE: usize = 120;
-    pub const CLONE_VM: usize = 0x100;
-    pub const CLONE_FS: usize = 0x200;
-    pub const CLONE_FILES: usize = 0x400;
-    pub const CLONE_VFORK: usize = 0x4000;
```
Comment (stratact) [inline kernel/syscall/common.rs]: Just a guess, I think @jackpot51 here explicitly had these constants indented to fit his style.

File: kernel/syscall/common.rs [hunk 11]
```diff
@@ -5,13 +5,13 @@ pub const SYS_BRK: usize = 45;
...
+pub const CLONE_VM: usize = 0x100;
+pub const CLONE_FS: usize = 0x200;
+pub const CLONE_FILES: usize = 0x400;
+pub const CLONE_VFORK: usize = 0x4000;
 pub const SYS_CLOCK_GETTIME: usize = 265;
-    pub const CLOCK_REALTIME: usize = 1;
-    pub const CLOCK_MONOTONIC: usize = 4;
```
Comment (stratact) [inline kernel/syscall/common.rs]: Same here.

File: kernel/syscall/common.rs [hunk 12]
```diff
@@ -23,24 +23,24 @@ pub const SYS_FTRUNCATE: usize = 93;
 pub const SYS_GETPID: usize = 20;
 pub const SYS_LINK: usize = 9;
 pub const SYS_LSEEK: usize = 19;
-    pub const SEEK_SET: usize = 0;
-    pub const SEEK_CUR: usize = 1;
-    pub const SEEK_END: usize = 2;
```
Comment (stratact) [inline kernel/syscall/common.rs]: and here.

File: kernel/syscall/common.rs [hunk 13]
```diff
@@ -23,24 +23,24 @@ pub const SYS_FTRUNCATE: usize = 93;
...
-    pub const O_RDWR: usize = 2;
-    pub const O_NONBLOCK: usize = 4;
-    pub const O_APPEND: usize = 8;
-    pub const O_SHLOCK: usize = 0x10;
-    pub const O_EXLOCK: usize = 0x20;
-    pub const O_ASYNC: usize = 0x40;
-    pub const O_FSYNC: usize = 0x80;
-    pub const O_CREAT: usize = 0x200;
-    pub const O_TRUNC: usize = 0x400;
-    pub const O_EXCL: usize = 0x800;
```
Comment (stratact) [inline kernel/syscall/common.rs]: and here.

File: filesystem/apps/shell/main.rs [hunk 14]
```diff
@@ -282,9 +282,9 @@ impl<'a> Command<'a> {
                     println!("URL: {:?}", file.path());
 
                     let string: String = args.iter()
-                        .skip(2)
-                        .fold(String::new(), |s, arg| s + " " + arg)
-                        + "\r\n\r\n";
+                                             .skip(2)
+                                             .fold(String::new(), |s, arg| s + " " + arg) +
+                                         "\r\n\r\n";
```
Comment (stratact) [inline filesystem/apps/shell/main.rs]: This line is not indented consistently and I think it would look better to have the `+` in front of the this expression.

File: filesystem/apps/sodium/exec.rs [hunk 15]
```diff
@@ -38,34 +37,34 @@ impl Editor {
...
+                    //                            self.goto(p);
+                    //                        }
+                    //                    },
+                    //                    Char('p') => {
+                    //                        let p = self.previous(n);
+                    //                        if let Some(p) = p {
+                    //                            self.goto(p);
+                    //                        }
+                    //                    },
+                    //                    // }
```
Comment (stratact) [inline filesystem/apps/sodium/exec.rs]: Could you remove the extra whitespace between the `//` and the expressions?

File: filesystem/apps/sodium/exec.rs [hunk 16]
```diff
@@ -189,21 +188,21 @@ impl Editor {
...
-//                    },
+                    }
+                    //
+                    //                    ????
+                    //                    Char('K') => {
+                    //                        self.goto((0, 0));
+                    //                    },
+                    //                    Char('J') => {
+                    //                        self.goto((0, self.text.len() - 1));
+                    //                    },
```
Comment (stratact) [inline filesystem/apps/sodium/exec.rs]: Same here.

File: filesystem/apps/editor/main.rs [hunk 17]
```diff
@@ -153,11 +161,8 @@ impl Editor {
     }
 
     fn main(&mut self, url: &str) {
-        let mut window = Window::new(-1,
-                                     -1,
-                                     576,
-                                     400,
-                                     &("Editor (".to_string() + url + ")")).unwrap();
```
Comment (skylerberg) [inline filesystem/apps/editor/main.rs]: This looks intentional to prevent the line from being overly long.

**Comment by Ticki** [comment]
@jackpot @stratact This patch is now fixed.

**Comment by iankronquist** [comment]
Does this PR remove the trailing `return` statements mentioned in the [TODO list](https://github.com/redox-os/redox/blame/master/TODO.md#L17)? If not, I'm willing to take a whack at that.

**Comment by Ticki** [comment]
The todo list is outdated. Will remove it.
